### PR TITLE
new func to add agencies

### PR DESCRIPTION
**Pull request by asuender**
I've added a function to add all appearing agencies and made some changes at `collectDepsAt()` and `addLine()`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -168,8 +178,10 @@ const generateGtfs = async (hafas, createWritable, serviceArea, opt = {}) => {
 			.sort((a, b) => new Date(a.plannedWhen) - new Date(b.plannedWhen))
 
 			for (const dep of deps) {
-				if (trips.has(dep.tripId)) continue
-				queue.push(fetchTrip(dep, station))
+				if (!trips.has(dep.tripId))
+					queue.push(fetchTrip(dep, station))
+				if (!agencies.has(dep.line.operator.name))
```
Comment (derhuerst) [inline index.js]: Please use `dep.line.operator.id` as the key in `agencies`.

File: index.js [hunk 2]
```diff
@@ -100,7 +108,9 @@ const generateGtfs = async (hafas, createWritable, serviceArea, opt = {}) => {
 		if (routes.has(line.id)) return;
 		routes.set(line.id, {
 			route_id: line.id,
-			// todo: agency_id
+			// leave this temporary blank, see e.g.
```
Comment (derhuerst) [inline index.js:111]: Please keep the todo here. Referencing the `agency.txt` entry from the `routes.txt` entry still needs to be done.

**Review by derhuerst — CHANGES_REQUESTED** [review]
Thanks for your contribution and sorry for taking so long to respond! I have some minor change requests.

**Comment by derhuerst** [comment]
@asuender ping 😉

**Comment by asuender** [comment]
Sorry I hadn‘t much time to fix this, I gonna do this soon. Sorry again.

**Comment by derhuerst** [comment]
@asuender another ping. if you don‘t have time, that‘s fine obviously

**Comment by derhuerst** [comment]
Thanks!

**Pull request merged**